vnext/Microsoft.ReactNative/Views/ReactRootControl.cpp

```diff
@@ -238,58 +241,70 @@ void ReactRootControl::UninitRootView() noexcept {
   m_isInitialized = false;
 }
 
-void ReactRootControl::ShowInstanceLoaded(Mso::React::IReactInstance &reactInstance) noexcept {
-  if (XamlView xamlRootView = m_weakXamlRootView.get()) {
-    auto xamlRootGrid{xamlRootView.as<winrt::Grid>()};
-
-    // Remove existing children from root view (from the hosted app)
-    xamlRootGrid.Children().Clear();
-
-    auto &legacyInstance = query_cast<Mso::React::ILegacyReactInstance &>(reactInstance);
-    legacyInstance.AttachMeasuredRootView(this, Mso::Copy(m_reactViewOptions->InitialProps));
-    m_isJSViewAttached = true;
-  }
-}
-
-void ReactRootControl::ShowInstanceError() noexcept {
+void ReactRootControl::ClearLoadingUI() noexcept {
   if (XamlView xamlRootView = m_weakXamlRootView.get()) {
     auto xamlRootGrid{xamlRootView.as<winrt::Grid>()};
 
-    // Remove existing children from root view (from the hosted app)
-    xamlRootGrid.Children().Clear();
+    auto children = xamlRootGrid.Children();
+    uint32_t index{0};
+    if (m_greenBoxGrid && children.IndexOf(m_greenBoxGrid, index)) {
+      children.RemoveAt(index);
+    }
   }
 }
 
-void ReactRootControl::ShowInstanceWaiting(Mso::React::IReactInstance & /*reactInstance*/) noexcept {
+void ReactRootControl::EnsureLoadingUI() noexcept {
   if (XamlView xamlRootView = m_weakXamlRootView.get()) {
     auto xamlRootGrid{xamlRootView.as<winrt::Grid>()};
 
-    // Remove existing children from root view (from the hosted app)
-    xamlRootGrid.Children().Clear();
-
     // Create Grid & TextBlock to hold text
     if (m_waitingTextBlock == nullptr) {
       m_waitingTextBlock = winrt::TextBlock();
       m_greenBoxGrid = winrt::Grid{};
       m_greenBoxGrid.Background(xaml::Media::SolidColorBrush(winrt::ColorHelper::FromArgb(0xff, 0x03, 0x59, 0)));
       m_greenBoxGrid.Children().Append(m_waitingTextBlock);
       m_greenBoxGrid.VerticalAlignment(xaml::VerticalAlignment::Center);
+
+      // Format TextBlock
+      m_waitingTextBlock.TextAlignment(winrt::TextAlignment::Center);
+      m_waitingTextBlock.TextWrapping(xaml::TextWrapping::Wrap);
+      m_waitingTextBlock.FontFamily(winrt::FontFamily(L"Consolas"));
+      m_waitingTextBlock.Foreground(xaml::Media::SolidColorBrush(winrt::Colors::White()));
+      winrt::Thickness margin = {10.0f, 10.0f, 10.0f, 10.0f};
+      m_waitingTextBlock.Margin(margin);
     }
 
-    // Add box grid to root view
-    xamlRootGrid.Children().Append(m_greenBoxGrid);
+    auto children = xamlRootGrid.Children();
+    uint32_t index;
```

Comment:
```suggestion
    uint32_t index{0};
```
Either that's needed here or it's not needed above.

Reply:
It shouldn't really be needed, since the value is only used if IndexOf returns true indicating that it set index.